lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -2,11 +2,30 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi
       class ResourceIdentifier
+        def self.type_for(class_name, serializer_type = nil, transform_options = {})
+          if serializer_type
+            raw_type = serializer_type
+          else
+            inflection =
+              if ActiveModelSerializers.config.jsonapi_resource_type == :singular
+                :singularize
+              else
+                :pluralize
+              end
+
+            raw_type = class_name.underscore
+            raw_type = ActiveSupport::Inflector.public_send(inflection, raw_type)
+            raw_type
+              .gsub!('/'.freeze, ActiveModelSerializers.config.jsonapi_namespace_separator)
```

Comment:
on another note, I'm concerned about coded cleanliness with inline `.freeze` Is there a way to turn on frozen strings by default for just AMS?